### PR TITLE
Aws pricing client testing

### PR DESCRIPTION
**Pull request by Miidoriya**
- Added lint check to github actions CI checks on `push` and `pull requests`
- updated API modularity and client usage
- formatting

### CODE REVIEW

File: src/services/aws/pricing/function/GetProducts.spec.ts [hunk 1]
```diff
@@ -1,11 +1,12 @@
-import { getProducts } from "./GetProducts"
+import { getProducts } from "./GetProducts";
 import { filtersMapMock, productsArrayMock } from "./WriteProductsToFile.mocks";
-import { client } from "../index"
-import  * as fs from "fs";
```
Comment (nia-vf) [inline src/services/aws/pricing/function/GetProducts.spec.ts:4]: No `fs` now? Was it being used before?

File: src/services/aws/pricing/function/GetProducts.ts [hunk 2]
```diff
@@ -20,28 +21,31 @@ const createFilters = (filtersMap: Map<string, string>) => {
   return filters;
 };
 
-export const getProducts = async (filtersMap: Map<string, string> ): Promise<Product[]> => {
+export const getProducts = async (
+  filtersMap: Map<string, string>,
+  priceClient: PricingClient = client
```
Comment (nia-vf) [inline src/services/aws/pricing/function/GetProducts.ts:26]: where is `client` variable coming from?

Comment (nia-vf) [reply src/services/aws/pricing/function/GetProducts.ts:26]: sorry just seen it is imported from client-pricing

File: src/services/aws/pricing/Pricing.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { PricingClient, PricingClientConfig } from "@aws-sdk/client-pricing";
+import { Product } from "../../../interfaces/product";
+import { getProducts } from "./function/GetProducts";
+import { writeProductsToFile } from "./function/WriteProductsToFile";
```
Comment (nia-vf) [inline src/services/aws/pricing/Pricing.ts:4]: Importing this function, but where is it used?

File: src/services/aws/pricing/function/WriteProductsToFile.mocks.ts [hunk 4]
```diff
@@ -1,565 +1,582 @@
...
-            "offerTermCode": "JRTCKXETXF",
-            "termAttributes": {}
-          }
-        }
+          },
+          sku: "Y6JK4U8NZRJW6Z2M",
+          effectiveDate: "2022-10-01T00:00:00Z",
+          offerTermCode: "JRTCKXETXF",
+          termAttributes: {},
+        },
```
Comment (nia-vf) [inline src/services/aws/pricing/function/WriteProductsToFile.mocks.ts:74]: why is there a comma following the closing bracket of the object?

Comment (nia-vf) [reply src/services/aws/pricing/function/WriteProductsToFile.mocks.ts:74]: Same thing on Line 73

Comment (nia-vf) [reply src/services/aws/pricing/function/WriteProductsToFile.mocks.ts:74]: And in other places

File: src/services/aws/pricing/Pricing.ts (hunk 3 above)
Comment (Miidoriya) [reply src/services/aws/pricing/Pricing.ts:4]: Will remove

File: src/services/aws/pricing/function/GetProducts.ts (hunk 2 above)
Comment (Miidoriya) [reply src/services/aws/pricing/function/GetProducts.ts:26]: 👍

File: src/services/aws/pricing/function/GetProducts.spec.ts (hunk 1 above)
Comment (Miidoriya) [reply src/services/aws/pricing/function/GetProducts.spec.ts:4]: Needed removal as it was unused

File: src/services/aws/pricing/function/WriteProductsToFile.mocks.ts (hunk 4 above)
Comment (Miidoriya) [reply src/services/aws/pricing/function/WriteProductsToFile.mocks.ts:74]: This is a snippet of pricing data returned from the pricing API unedited. The comma at the end has no effect, it's more a convention to have or have not but it shouldn't be inconsistent as it is in this file. As it's just a mock we can leave it as is

**Review by imarns — APPROVED** [review]

**Pull request merged**